api/src/main/java/io/opentelemetry/api/internal/ImmutableKeyValuePairs.java

```diff
@@ -3,34 +3,36 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-package io.opentelemetry.api.common;
+package io.opentelemetry.api.internal;
 
 import static io.opentelemetry.api.internal.Utils.checkArgument;
 
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.Labels;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 import javax.annotation.Nullable;
 import javax.annotation.concurrent.Immutable;
 
 /**
- * An immutable set of key-value pairs. Keys are only {@link String} typed.
+ * An immutable set of key-value pairs.
  *
  * <p>Key-value pairs are dropped for {@code null} or empty keys.
  *
  * <p>Note: for subclasses of this, null keys will be removed, but if your key has another concept
- * of being "empty", you'll need to remove them before calling {@link #sortAndFilter(Object[])},
- * assuming you don't want the "empty" keys to be kept in your collection.
+ * of being "empty", you'll need to remove them before calling {@link #sortAndFilter(Object[],
+ * boolean)}, assuming you don't want the "empty" keys to be kept in your collection.
  *
  * @param <V> The type of the values contained in this.
  * @see Labels
  * @see Attributes
  */
 @SuppressWarnings("rawtypes")
 @Immutable
-abstract class ImmutableKeyValuePairs<K, V> {
+public abstract class ImmutableKeyValuePairs<K, V> {
 
-  List<Object> data() {
+  public List<Object> data() {
```

Comment:
this does bum me out quite a bit. By making this public, we've now exposed implementation details on all of Attributes, Labels and Baggage that could easily be abused. Can we make this protected, so the subclasses get access, but the public does not?

Reply:
Just tried it locally, and yes, we can make this protected. I'll feel much better about things then.

Reply:
and also all the subclasses can likewise have their overrides made protected.

Reply:
Sweet thanks!